larpandoracontent/LArControlFlow/NeutrinoIdTool.cc

```diff
@@ -158,21 +159,21 @@ void NeutrinoIdTool::Collect2DHits(const PfoList &pfos, CaloHitList &hitList, Ca
         // ATTN hits collected from Pfos are copies of hits passed from master instance, we need to access their parent to use MC info
         const CaloHit *pParentCaloHit(static_cast<const CaloHit *>(pCaloHit->GetParentAddress()));
 
-        if (std::find(reconstructableCaloHitList.begin(), reconstructableCaloHitList.end(), pParentCaloHit) == reconstructableCaloHitList.end())
+        if (!reconstructableCaloHitSet.count(pParentCaloHit))
             continue;
 
         // Ensure no hits have been double counted
-        if (std::find(hitList.begin(), hitList.end(), pParentCaloHit) == hitList.end())
-            hitList.push_back(pParentCaloHit);
+        if (!reconstructedCaloHitSet.count(pParentCaloHit))
+            reconstructedCaloHitSet.insert(pParentCaloHit);
     }
 }
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
-unsigned int NeutrinoIdTool::CountNeutrinoInducedHits(const CaloHitList &hitList) const
+unsigned int NeutrinoIdTool::CountNeutrinoInducedHits(const CaloHitSet &caloHitSet) const
 {
     unsigned int nNuHits(0);
-    for (const CaloHit *const pCaloHit : hitList)
+    for (const CaloHit *const pCaloHit : caloHitSet)
     {
```

Comment:
As discussed, use unordered_set for the look-ups, but list (or vector) for ordered loops (making sure that contents of list of vector hadn't been in a set as n intermediate step!).